File: test_rollover_ui.py
import pytest
import os
import time
from playwright.sync_api import sync_playwright

def test_rollover_ui():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()

        # Add console listener BEFORE goto
        page.on("console", lambda msg: print(f"Browser Console: {msg.text}"))
        page.on("pageerror", lambda err: print(f"Browser Error: {err}"))

        page.goto("http://localhost:8000/workbench", timeout=60000)

        # Click on derivatives analysis tab
        page.evaluate("switchMainTab('derivatives')")
        page.wait_for_timeout(1000)

        # Click on Rollover Analysis sub-tab
        page.evaluate("switchDerivTab('rollover')")
        page.wait_for_timeout(5000)  # Wait longer for data

        # Take screenshot
        page.screenshot(path="rollover_ui_check.png")

        browser.close()

if __name__ == "__main__":
    test_rollover_ui()

File: test_rollover_ui.js
const { chromium } = require('playwright');

(async () => {
    const browser = await chromium.launch({ headless: true });
    const context = await browser.newContext({
        viewport: { width: 1920, height: 1080 }
    });
    const page = await context.newPage();

    try {
        await page.goto('http://127.0.0.1:8000/workbench');

        await page.waitForTimeout(4000);

        await page.click('text=Derivatives Analysis');
        await page.waitForTimeout(2000);
        await page.click('text=Rollover Analysis');
        await page.waitForTimeout(2000);

        // Click master sync at the top right to populate data
        const syncButton = await page.locator('button:has-text("Master Sync")');
        if (await syncButton.isVisible()) {
            await syncButton.click();
            await page.waitForTimeout(15000); // wait for global sync to fetch data
        } else {
             console.log("Master Sync not found");
        }

        await page.screenshot({ path: '/home/jules/verification/rollover_styled_5.png', fullPage: true });
        console.log('Successfully captured screenshot of Rollover Analysis with styling.');

    } catch (e) {
        console.error('Test failed:', e);
    } finally {
        await context.close();
        await browser.close();
    }
})();
